SpecFlowProject4/StepDefinitions/DynamicControlsStepDefinitions.cs

```diff
@@ -0,0 +1,48 @@
+﻿using ApiTests.Tests.UI;
+using TestProject.Tests.Pages;
+using TestProject.Utils;
+using NUnit.Framework;
+using SpecFlowProject4.StepDefinitions;
+
+namespace DynamicControlsStepDefinitions
+{
+    [Binding]
+    internal class DynamicControlsStepDefinitions : SetUp
+    {
+        readonly DynamicControlsPage dynamicControlsPage = new();
+        readonly BaseTest baseTest = new();
+        public DynamicControlsStepDefinitions(ScenarioContext _scenarioContext) : base(_scenarioContext)
+        {
+        }
+
+        [Given(@"I Click '(.*)' button")]
+        public void GivenIClickButton(string Enable)
+        {
+            baseTest.ClickOnElement(Enable);
+            dynamicControlsPage.WaitForDisableBtn();
+        }
+
+        [Then(@"Enable/disable input is enabled")]
```

Comment:
I'd also suggest to try out step argument transformations, they will be a great help with this type of steps